de.fu_berlin.inf.dpp.server/src/de/fu_berlin/inf/dpp/server/ServerLifecycle.java

```diff
@@ -0,0 +1,78 @@
+package de.fu_berlin.inf.dpp.server;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+
+import org.apache.log4j.Logger;
+
+import de.fu_berlin.inf.dpp.account.XMPPAccount;
+import de.fu_berlin.inf.dpp.account.XMPPAccountStore;
+import de.fu_berlin.inf.dpp.communication.connection.ConnectionHandler;
+import de.fu_berlin.inf.dpp.context.AbstractContextLifecycle;
+import de.fu_berlin.inf.dpp.context.ContainerContext;
+import de.fu_berlin.inf.dpp.context.IContextFactory;
+import de.fu_berlin.inf.dpp.filesystem.IProject;
+import de.fu_berlin.inf.dpp.filesystem.IResource;
+import de.fu_berlin.inf.dpp.net.xmpp.JID;
+import de.fu_berlin.inf.dpp.session.ISarosSessionManager;
+import de.fu_berlin.inf.dpp.session.SessionEndReason;
+
+public class ServerLifecycle extends AbstractContextLifecycle {
+
+    private static final Logger log = Logger.getLogger(ServerLifecycle.class);
+    
+    @Override
+    protected Collection<IContextFactory> additionalContextFactories() {
+        List<IContextFactory> factories = new ArrayList<IContextFactory>();
+        factories.add(new ServerContextFactory());
+        return factories;
+    }
+
+    @Override
+    protected void initializeContext(final ContainerContext context) {
+        connectToXMPPServer(context);
+        context.getComponent(ISarosSessionManager.class).startSession(new HashMap<IProject, List<IResource>>());
+    }
+
+    @Override
+    protected void finalizeContext(final ContainerContext context) {
+        context.getComponent(ISarosSessionManager.class).stopSession(
+            SessionEndReason.LOCAL_USER_LEFT);
+
+        context.getComponent(ConnectionHandler.class).disconnect();
+    }
+
+    /*
+    * FIXME This is currently ALPHA, the server always assumes it can connect to
+    * the XMPP server and never gets disconnected. Of course this is unrealistic !
```

Comment:
You can install some buggish SMACK staff but I would recommend to write your own listener.